### PR TITLE
Hotfix Update baninfo with error handling and more thorough unit test.

### PR DESCRIPTION
**Pull request by y4ure**

### CODE REVIEW

File: driftbase/flexmatch.py [hunk 1]
```diff
@@ -258,11 +258,13 @@ def handle_match_event(queue_name, event_data):
...
-            with BanInfo(player_id=event_data["player_id"], match_type=event_data["match_type"]) as ban_info:
-                ban_info.update_ban(event_data["match_id"])
+            player_id, match_id, match_type = event_data["player_id"], event_data["match_id"], event_data["match_type"]
+            log.info(f"Player {player_id} in {match_type} match {match_id} banned.")
+            with BanInfo(player_id=player_id, match_type=match_type) as ban_info:
+                ban_info.update_ban(match_id)
                 if ban_info.exists():
-                    player_id = event_data["player_id"]
-                    log.info(f"Player {player_id} banned from matchmaking: {ban_info.matchmaker} "
+                    log.info(f"Player {player_id} ban info updated: "
```
Comment (judgeaxl) [inline driftbase/flexmatch.py:266]: If you don't want to get very long log lines that are hard to parse and search it's useful to use the "extra" parameter, and pass a dict which will be included as meta data, and more easily searchable. So you can include values in the log message which have few variations, but put values with a large number of variations in extra. E.g. player-id goes in extra, but match type could go in the message. Normally you want to get an understanding about what's going on by just browsing the log, and all the details go in extra.

Comment (y4ure) [reply driftbase/flexmatch.py]: Updated.
